### PR TITLE
F_t add factor nonnegativity checks for adjust_*

### PR DESCRIPTION
**Pull request by SsnL**

### CODE REVIEW

File: torchvision/transforms/functional_tensor.py [hunk 1]
```diff
@@ -281,8 +290,10 @@ def ten_crop(img, size, vertical_flip=False):
 
 def _blend(img1, img2, ratio):
     # type: (Tensor, Tensor, float) -> Tensor
-    bound = 1 if img1.dtype in [torch.half, torch.float32, torch.float64] else 255
-    return (ratio * img1 + (1 - ratio) * img2).clamp(0, bound).to(img1.dtype)
+    if img1.dtype in [torch.half, torch.float32, torch.float64]:
+        return torch.lerp(ratio, img2, img1).clamp(0, 1).to(img1.dtype)
+    else:
+        return torch.lerp(ratio, img2.float(), img1.float()).clamp(0, 255).to(img1.dtype)
```
Comment (fmassa) [inline torchvision/transforms/functional_tensor.py:296]: Looks like torchscript might not support this overload yet. Can you revert those changes?

Comment (SsnL) [reply torchvision/transforms/functional_tensor.py]: updated!

**Comment by fmassa** [comment]
Test failures seems related, can you fix those?

File: torchvision/transforms/functional_tensor.py [hunk 2]
```diff
@@ -110,10 +113,13 @@ def adjust_contrast(img, contrast_factor):
...
         Tensor: Contrast adjusted image.
     """
+    if contrast_factor < 0:
+        raise ValueError('contrast_factor ({}) is not non-negative.'.format(contrast_factor))
+
     if not _is_tensor_a_torch_image(img):
         raise TypeError('tensor is not a torch image.')
 
-    mean = torch.mean(rgb_to_grayscale(img).to(torch.float))
+    mean = rgb_to_grayscale(img).mean()
```
Comment (fmassa) [inline torchvision/transforms/functional_tensor.py:122]: mean is currently not supported for `uint8` types. I think this will be soon fixed in PyTorch, but in the meantime, can you revert this change as well?

Comment (SsnL) [reply torchvision/transforms/functional_tensor.py]: reverted!
